source/extensions/quic_listeners/quiche/spdy_server_push_utils_for_envoy.cc

```diff
@@ -0,0 +1,35 @@
+#include "quiche/quic/core/http/spdy_server_push_utils.h"
+
+// NOLINT(namespace-envoy)
+
+// This file has a substitute definition for
+// quiche/quic/core/http/spdy_server_push_utils.cc which depends on GURL.
+// Since Envoy doesn't support server push, these functions shouldn't be
+// executed at all.
+
+using spdy::SpdyHeaderBlock;
+
+namespace quic {
+
+// static
+std::string SpdyServerPushUtils::GetPromisedUrlFromHeaders(const SpdyHeaderBlock& /*headers*/){
```

Comment:
Should we use `NOT_IMPLEMENTED_GCOVR_EXCL_LINE` PANIC instead throughout this file?

Reply:
Yeah, NOT_IMPLEMENTED_GCOVR_EXCL_LINE seems more suitable here. Done